src/node/rpc/node_frontend.h

```diff
@@ -461,9 +465,24 @@ namespace ccf
           q.format = node_quote_info.format;
 
 #ifdef GET_QUOTE
-          auto code_id =
-            EnclaveAttestationProvider::get_code_id(node_quote_info);
-          q.mrenclave = ds::to_hex(code_id.data);
+          // get_code_id attempts to re-validate the quote to extract mrenclave
+          // and the Open Enclave is insufficiently flexible to allow quotes
+          // with expired collateral to be parsed at all. Recent nodes therefore
+          // cache their code digest on startup, and this code attempts to fetch
+          // that value when possible and only call the unreliable get_code_id
+          // otherwise.
+          auto nodes = args.tx.ro(network.nodes);
+          auto node_info = nodes->get(context.get_node_state().get_node_id());
+          if (node_info.has_value() && node_info->code_digest.has_value())
+          {
+            q.mrenclave = node_info->code_digest.value();
+          }
+          else
+          {
+            auto code_id =
+              EnclaveAttestationProvider::get_code_id(node_quote_info);
```

Comment:
In the current case where this fails, it throws. Should we catch that and turn it into a nicer error?

Reply:
We could do, but this isn't the exact endpoint logic ACL will use anyway, so I think we should discuss it with them and find out what they prefer.